hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -897,6 +911,20 @@ void ensureBlockReady() throws IOException {
       finishBlock();
     }
 
+    public boolean shouldFinishBlock() throws IOException {
+//      int uncompressedBlockSize = blockSizeWritten();
+      int uncompressedBlockSize = baosInMemory.size();
+      if (uncompressedBlockSize >= fileContext.getBlocksize()) {
+        if (uncompressedBlockSize < maxSizeUnCompressed) {
+          int adjustedBlockSize = compressedSizePredicator.
+            calculateCompressionSizeLimit(fileContext, uncompressedBlockSize);
+          return uncompressedBlockSize >= adjustedBlockSize;
+        }
+        return true;
+      }
+      return false;
+    }
+
```

Comment:
I'll also move this to Predicator and generalize this method 
```suggestion
    public boolean checkBoundariesWithPredicate() throws IOException {
    if(predicator==null){
    throw new IllegalArgumentException("Expected at least the default BoundariesCheckPredicate");
    }
         return predicator.
            shouldFinishBlock(fileContext, uncompressedBlockSize);
        
        
    }

```